src/transform/src/column_knowledge.rs

```diff
@@ -242,8 +242,12 @@ impl ColumnKnowledge {
                             // These methods propagate constant values exactly.
                             knowledge
                         }
+                        AggregateFunc::Count => DatumKnowledge {
```

Comment:
This is need only for completeness. Both transforms overlap, ColumnKnowledge wasn't detecting `COUNT(..) is null` as a `false` predicate, but NonNullRequirements was.

Reply:
Issue #6613 is about eliminating this overlap. I prefer eliminating the nonnull parts of `ColumnKnowledge` myself.